packages/block-editor/src/components/inserter-list-item/style.scss

```diff
@@ -49,6 +49,8 @@
 				border-radius: $radius-block-ui;
 				opacity: 0.04;
 				background: var(--wp-admin-theme-color);
+				// This fixes drag-and-drop in Firefox.
+				pointer-events: none;
```

Comment:
I think we can keep this here since the added tests caught this bug. If we separate this into another PR then it wouldn't make the full picture (the corresponding test is in another PR).